roles/moving_objects_between_buckets/tasks/main.yml

```
@@ -0,0 +1,28 @@
+---
+- name: Run 'moving_object_between_buckets' role
+  module_defaults:
+    group/aws: "{{ aws_setup_credentials__output }}"
+
+  block:
+    - name: Obtaining all the source bucket objects
+      amazon.aws.s3_object:
+        bucket: "{{ moving_bucket_src }}"
+        mode: list
+      register: bucket_src_info
+
+    - name: Moving objects between source bucket to destination bucket
+      amazon.aws.s3_object:
+        bucket: "{{ moving_bucket_dest }}"
+        object: "{{ item }}"
+        mode: copy
+        copy_src:
+            bucket: "{{ moving_bucket_src }}"
+            object: "{{ item }}"
+      loop: "{{ bucket_src_info.s3_keys }}"
+
+    - name: Delete the source bucket objects
+      amazon.aws.s3_object:
+        bucket: "{{ moving_bucket_src }}"
+        object: "{{ item }}"
+        mode: delobj 
+      loop: "{{ bucket_src_info.s3_keys }}"
```

newline at end of file
```suggestion
      loop: "{{ bucket_src_info.s3_keys }}"

```